### DISCUSSION TITLE
How to respawn bot?

### DISCUSSION THREAD

**Discussion by fungoza** [PrismarineJS/mineflayer#3014] [Q&A]
Server has anti-cheat that prevent respawning in 2 sec after death
How to respawn bot manually?

**Comment by SonicandTailsCD** [comment]
You can't (at least that I know of) :P
Try to add a way for when your bot dies, your bot leaves and rejoins... that can help :)

**Comment by SonicandTailsCD** [comment]
I suggest this: 
```js
bot.on('entityGone', async (entity) => {
    if (entity === bot.username) {
    // when the bot dies   
    bot.quit()
    // quit the game
    }
}) 
bot.on('end', createBot)
createBot()
```
You will need to move your bot spawning constant to a function first.

**Reply by fungoza** [reply]
Server doesn't access respawn after rejoin anyway

**Reply by SonicandTailsCD** [reply]
> Server doesn't access respawn after rejoin anyway

What do you mean?

**Comment by amoraschi** [comment]
The previous code is comparing an Entity instance with a string.

I think there was a PR that disabled automatic respawn, but it hasn't had any activity in a while, you could look at it and perhaps fix the PR.

**Reply by SonicandTailsCD** [reply]
Oh, sorry :(

**Comment by fungoza** [comment]
I see few people here, maybe I need to create an issue with a request so that there are more people

**Reply by SonicandTailsCD** [reply]
Yeah, you need to do that. :)
